lib/test.js

```diff
@@ -537,7 +538,7 @@ Test.prototype['throws'] = function (fn, expected, msg, extra) {
         fn();
     } catch (err) {
         caught = { error: err };
-        if (err != null && (!isEnumerable(err, 'message') || !has(err, 'message'))) {
+        if (typeof err === 'object' && err != null && (!isEnumerable(err, 'message') || !has(err, 'message'))) {
```

Comment:
This won't let a function through, altho it'd be weird to throw a function. i'll tweak the check a bit.